R/RelatedRecords.R

```diff
@@ -0,0 +1,141 @@
+#' @title RelatedRecords
+#'
+#' @description
+#' A container for accessing records with a one-to-many or many-to-many
+#' relationship.
+RelatedRecords <- R6::R6Class( # nolint object_name_linter
+  "RelatedRecords",
+  cloneable = FALSE,
+  public = list(
+    #' @description
+    #' Creates an instance of this R6 class. This class should not be instantiated directly,
+    #' but rather by connecting to a LaminDB instance using the [connect()] function.
+    #'
+    #' @param instance The instance the records list belongs to.
+    #' @param registry The registry the records list belongs to.
+    #' @param field The field associated with the records list.
+    #' @param related_to ID or UID of the parent that records are related to.
+    #' @param api The API for the instance.
+    initialize = function(instance, registry, field, related_to, api) {
+      private$.instance <- instance
+      private$.registry <- registry
+      private$.api <- api
+      private$.field <- field
+      private$.related_to <- related_to
+    },
+    #' @description
+    #' Get a data frame summarising records in the registry
+    #'
+    #' @param limit Maximum number of records to return
+    #' @param verbose Boolean, whether to print progress messages
+    #'
+    #' @return A data.frame containing the available records
+    df = function(limit = 100, verbose = FALSE) {
+      private$get_records(as_df = TRUE)
+    },
+    #' @description
+    #' Print a `RelatedRecords`
+    #'
+    #' @param style Logical, whether the output is styled using ANSI codes
+    print = function(style = TRUE) {
+      cli::cat_line(self$to_string(style))
+    },
+    #' @description
+    #' Create a string representation of a `RelatedRecords`
+    #'
+    #' @param style Logical, whether the output is styled using ANSI codes
+    #'
+    #' @return A `cli::cli_ansi_string` if `style = TRUE` or a character vector
+    to_string = function(style = FALSE) {
+      fields <- list(
+        field_name = private$.field$field_name,
+        relation_type = private$.field$relation_type,
+        related_to = private$.related_to
+      )
+
+      field_strings <- make_key_value_strings(fields)
+
+      make_class_string(
+        "RelatedRecords", field_strings,
+        style = style
+      )
+    }
+  ),
+  active = list(
+    #' @field field ([Field])\cr
+    #' The field the records are related to.
+    field = function() {
+      private$.field
+    }
+  ),
```

Comment:
Why expose this interface? Does the Python interface also expose this information?